src/legacy/ui/public/index_patterns/_pattern_cache.ts

```diff
@@ -17,35 +17,33 @@
  * under the License.
  */
 
-export function createIndexPatternCache() {
-
-  const vals = {};
-  const cache = {};
-
-  const validId = function (id) {
-    return typeof id !== 'object';
-  };
-
-  cache.get = function (id) {
-    if (validId(id)) return vals[id];
-  };
-
-  cache.set = function (id, prom) {
-    if (validId(id)) vals[id] = prom;
-    return prom;
-  };
-
-  cache.clear = cache.delete = function (id) {
-    if (validId(id)) delete vals[id];
-  };
+export interface PatternCache {
+  get: (id: string) => any;
+  set: (id: string, value: any) => any;
+  clear: (id: string) => void;
+  clearAll: () => void;
+}
 
-  cache.clearAll = function () {
-    for (const id in vals) {
-      if (vals.hasOwnProperty(id)) {
-        delete vals[id];
+export function createIndexPatternCache(): PatternCache {
+  const vals: Record<string, any> = {};
+  const cache: PatternCache = {
+    get: (id: string) => {
+      return vals[id];
+    },
+    set: (id: string, prom: any) => {
```

Comment:
turns out it is actually a promise, since `IndexPatterns.make()` returns `IndexPattern.init()` which returns a promise. guessing it was named this way for clarity.

i've updated the `PatternCache` interface for clarity